app/src/main/java/com/passtick/test/ui/account/AccountScreen.kt

```diff
@@ -86,22 +92,90 @@ internal fun AccountScreen(
             }
         }
     ) {
-        Column {
-            if (openAddDialogue.value) {
-                AddAccountDialogue(
-                    openAddDialogue = openAddDialogue,
-                    listCoroutineScope = listCoroutineScope,
-                    listState = listState
-                )
+        if (openAddDialogue.value) {
+            AddAccountDialogue(
+                openAddDialogue = openAddDialogue,
+                listCoroutineScope = listCoroutineScope,
+                listState = listState,
+                accountList,
+            )
+        }
+
+        LazyColumn(state = listState) {
+            item {
+                Row(modifier) {
+                    SearchView(
+                        viewModel.textState,
+                        accountList,
+                        //filteredAccounts,
+                    )
+                }
+            }
+            items(items = viewModel.filteredAccounts.value, key = { it.uid}){ account ->
+                AccountDisplay(account = account)
             }
-            PasswordListDisplay(accountList = accountList, state = listState)
         }
     }
 
 }
 
-
-
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun SearchView(
+    textState: MutableState<TextFieldValue>,
+    accountList: List<Account>,
+) {
+    var viewModel: AccountViewModel = hiltViewModel()
+    TextField(
+        value = textState.value,
+        onValueChange = { value ->
+            textState.value = value
+            if(textState.value.text == "") {
+                viewModel.init(accountList)
+            } else {
+                viewModel.init(accountList.filter { it.serviceName == textState.value.text})
```

Comment:
This kind of works but only returns exact matches from the search bar. Should instead swap the flow with one generated by the function getAccountByServiceName() with the search text passed in as a parameter. This would be a more proper implementation of a search function.